packages/shared/src/loadClerkJsScript.ts

```diff
@@ -1,28 +1,53 @@
-import { parsePublishableKey } from '@clerk/shared/keys';
-import { loadScript } from '@clerk/shared/loadScript';
-import { isValidProxyUrl, proxyUrlToAbsoluteURL } from '@clerk/shared/proxy';
-import { addClerkPrefix } from '@clerk/shared/url';
-
-import { errorThrower } from '../errors/errorThrower';
-import type { IsomorphicClerkOptions } from '../types';
-import { isDevOrStagingUrl } from './isDevOrStageUrl';
+import type { ClerkOptions, SDKMetadata, Without } from '@clerk/types';
+
+import { buildErrorThrower } from './error';
+import { createDevOrStagingUrlCache, parsePublishableKey } from './keys';
+import { loadScript } from './loadScript';
+import { isValidProxyUrl, proxyUrlToAbsoluteURL } from './proxy';
+import { addClerkPrefix } from './url';
 import { versionSelector } from './versionSelector';
 
 const FAILED_TO_LOAD_ERROR = 'Clerk: Failed to load Clerk';
 
-type LoadClerkJsScriptOptions = Omit<IsomorphicClerkOptions, 'proxyUrl' | 'domain'> & {
+const { isDevOrStagingUrl } = createDevOrStagingUrlCache();
+
+const errorThrower = buildErrorThrower({ packageName: '@clerk/shared' });
+
+/**
+ * Sets the package name for error messages during ClerkJS script loading.
+ *
+ * @example
+ * setClerkJsLoadingErrorPackageName('@clerk/clerk-react');
+ */
+export function setClerkJsLoadingErrorPackageName(packageName: string) {
+  errorThrower.setPackageName({ packageName });
+}
+
+type LoadClerkJsScriptOptions = Without<ClerkOptions, 'isSatellite'> & {
+  publishableKey: string;
+  clerkJSUrl?: string;
+  clerkJSVariant?: 'headless' | '';
+  clerkJSVersion?: string;
+  sdkMetadata?: SDKMetadata;
   proxyUrl?: string;
   domain?: string;
   nonce?: string;
 };
 
-const loadClerkJsScript = (opts: LoadClerkJsScriptOptions) => {
-  const { publishableKey, nonce } = opts;
```

Comment:
A bad rebase is removing all `nonce` stuff in this file currently (added in https://github.com/clerk/javascript/pull/3858)

Reply:
Thanks Lennart, added